### ISSUE TITLE
Cannot spawn vehicle

### ISSUE THREAD

**Issue by kiprock** [matlabbe/rtabmap_drone_example#1]
Hello, I carefully followed your instructions, but running the 1st launch file roslaunch mavros_nav_2d gazebo.launch, the vehicle will not spawn, and i get the following message: 

`Resource not found: The following package was not found in <arg default="$(find mavlink_sitl_gazebo)/worlds/empty.world" name="world"/>: mavlink_sitl_gazebo`

I think my system is looking for mavlink_sitl_gazebo instead of posix_sitl_default. I am unsure of how to resolve this. Also, is it really necessary to use PX4 1.8.2? Why? 

Thank you

**Comment by matlabbe** [comment]
This example is not maintained unfortunately, so the PX4 1.8.2 was the version working at that time. I know Firmware 2 is out now, but the px4/mavros config of this project has to be migrated. At least https://github.com/matlabbe/mavros_nav_2d/blob/master/launch/slam.launch should be pretty similar for the latest px4 version. You may restart from latest px4/mavros drone example, and add the realsense camera like this to simulated drone: https://github.com/matlabbe/mavros_nav_2d/blob/356ba6e1d74a7fa4a5fcf5c2fd958c4be4dc10c2/urdf/drone.urdf.xacro#L276-L290

**Comment by matlabbe** [comment]
I updated the config file to work with PX4 v1.11.3, the current version I am using for another project that is yet to be finished (we tend to stick on same version for the whole project). I updated the README with installation instructions. I think the original [error](https://github.com/matlabbe/mavros_nav_2d/issues/1#issue-1020456718) was that git submodules of PX4 were not cloned.

When changing PX4 versions, make sure to cleanup the `eeprom` and `etc` folders created in `~/.ros`, I had some problems difficult to debug that may be caused by incompatible config files with different versions still in my `~/.ros` folder. `px4fmu_common` was crashing everytime.
